docs/proposals/20220901-add-edge-autonomy-e2e-tests.md

```diff
@@ -0,0 +1,83 @@
+In this work, we complete the e2e-tests of edge-nodes-autonomy for Openyurt by adjusting e2e testing framework, adding e2e-tests for related modules, which are yurthub, yurt-tunnel, kubelet, kube-proxy, coredns, flannel.
+
+## Motivation
+
+Openyurt provides ability of edge-nodes-autonomy of large-scale kubernetes clusters under offline scenarios. However,  the edge-nodes-autonomy ability has not been fully tested, which may cause unexpected problems. This propalsal intended to add e2e-tests for edge-nodes-autonomy ability.
+
+### Goals
+
+- Adjust e2e-test framework. Provide testing entrance for e2e-tests packages, including yurt, yurt-tunnel, yurthub, in the directory of /test/e2e/.
+- Adding e2e-tests for edge-node-autonomy related Openyurt modules, including yurthub, yurt-tunnel; vanilla kubernetes modules for offline situations, including kubelet, kube-proxy, coredns, flannel.
+- Modify e2e-test shell, to provide environment preparation and e2e-tests ochestration for modules mentioned above.
+- Removing e2e.go & e2e_test.go which currently locate in /test/e2e/.
+### Non-Goals
+
+- Basic module edge-autonomy-ability tests, which unit-tests are responsible for;
+- Non-edge-autonomy scenario related tests.
+
+## Proposal
+
+### User Stories
+
+#### Story 1
+As a developer of Openyurt, I want to get instant e2e-test-feedback after I made changes to Openyurt modules which may affect edge-node-autonomy ability and  commit the changes.
+
+#### Story 2
+As a user of Openyurt,  I want to make it clear, when I debug, whether it's the Openyurt edge-autonomy-modules are designed with problems, or it's other problems such as something wrong with my kubeadm cluster.
+
+### Implementation Details
+- Ajusting e2e-tests framework
+  - Bump ginkgo to v2 (Done,  pr #945);
+  - Create test-entrance in yurt.go, yurt-tunnel.go, yurthub.go in /test/e2e/;
+  - Add edge-autonomy labels to edge-autonomy related e2e-tests.
+- Adding e2e-tests for modules
+  - Kubelet
+    - Run a busybox pod(which does not need cache) on each edge node when they're still online;
+    - Check if the pod has successfully restarted after disconnection from cloud node. If so, kubelet is working normally.
+  - Yurthub
+    - After disconnection, use the cat/curl command on 127.0.0.1:10267/v1/healthz to see if yurthub  successfully restarts and is in healthy running status.
+  - Kube-proxy
+    - After disconnection, kill the kube-proxy pod, and request to ping other nodes. If the requests succeed before timeout, kube-proxy works. This requires there are services exposing instances on each edge node before the requests are made.
+  - Flannel
+    - After disconnection, ping other edge nodes using cluster ip. If it works, flannel had restarted and functioned successfully.
```

Comment:
We should first be in a pod and exec command to ping another pod on another node with PodIP.
Additionally, before test, we have to restart flannel and these pods on both nodes. The restart sequence is matter, we should first restart flannel and then these test pods. It helps us to test IPAM function of cni.